src/app/availability/availability-table/availability-table.component.ts

```diff
@@ -0,0 +1,56 @@
+﻿import { Component, OnInit } from '@angular/core';
+import { AvailabilityService } from '../availability.service';
+import { Location } from '@angular/common';
+import { AvailabilityDto } from '../../declarations/models/availability-dto';
+import { SprintService } from '../../common/sprint.service';
+import { SprintStateDto } from '../../declarations/models/sprint-state-dto';
+import { SprintDto } from '../../declarations/models/sprint-dto';
+import { ProjectDto } from '../../declarations/models/project-dto';
+import { MessageService } from 'primeng/api';
+
+@Component({
+  selector: 'app-availability-table',
+  templateUrl: './availability-table.component.html',
+  styleUrls: [ './availability-table.component.css' ]
+})
+export class AvailabilityTableComponent implements OnInit {
+
+  availability: AvailabilityDto;
+
+  declarableSprint: SprintDto;
+
+  // TODO move to store
+  get currentProject(): ProjectDto {
+    return { id: 1, initialCoefficient: 1.1, name: 'xx', description: 'xx' };
+  }
+
+  constructor(
+    private sprintService: SprintService,
+    private availabilityService: AvailabilityService,
+    private messageService: MessageService,
+    private location: Location) {
+  }
+
+  ngOnInit() {
+    this.sprintService.getSprints(this.currentProject.id, [ 'declarable' ]).subscribe(sprints => {
```

Comment:
Ja niekoniecznie wiem, gdzie dzieje się ta magia, ale Java tutaj chce `DECLARABLE` wielkimi literami generalnie:

```
java.lang.IllegalArgumentException: No enum constant pl.edu.agh.papaya.api.model.SprintStateDto.declarable
```

Reply:
Jeśli to jest jakieś problematyczne czy coś, to można te enumy w yamlu zmienić na takie wielkimi literami

Reply:
Bo w Javie to jest `SprintState.UPCOMING` ze stringiem `"upcoming"` w środku

Reply:
I z magii generacji na froncie powstało `'declarable`', ale z moich profesjonalnych testów wynika, że jak w yamlu jest wielkimi literami, to na froncie zaskakująco też i wtedy się nie sypie

Reply:
To się już powinno ogarniać z backendem z issue 66?

Reply:
Bo poprawiłam sobie to `SprintState.DECLARABLE` (w yamlu w submodule na froncie :stuck_out_tongue:) i Hibernate robi jakieś dzikie selecty, ale front mi mówi: `"Obecnie nie ma żadnego sprintu, dla którego możesz podać swoją dostępność"` - co jest kłamstwem, bo ja mam zapisy na sprint od 20 dzisiaj do 23 w poniedziałek w projekcie **xsdcfvgbnh**

Reply:
Chyba trzeba będzie enumy na capslocki zmienić wszędzie
A co do _kłamstwa_, to czy ten projekt ma na pewno id=1? xd pewnie nie xd

Reply:
```
availability-table.component.ts -> currentProject -> id
```

Reply:
Było blisko, bo miał id=2

Reply:
Dostał id=1 i jesteśmy tu:
![obraz](https://user-images.githubusercontent.com/34098778/57164820-8fe3d900-6df5-11e9-8e2d-c54ce69c82d9.png)

Reply:
okeeeej, a co masz w konsoli?
zrobiłem mały fix - projekt który się wybierze w `/projects` jest trzymany w prowizorycznym storze + wysyłam DECLARABLE

Reply:
i jeszcze build failuje, ach

Reply:
wydaje mi się, że takie coś jest możliwe jeśli AvailabilityDto przyjdzie jako null (albo raczej nie przyjdzie w ogóle)

Reply:
Build failuje przez `DECLARABLE`, nie? Ja mam to profesjonalnie poprawione u siebie, także spoko

Reply:
A no może nie przyjść

Reply:
W sensie
```
{
    "timestamp": "2019-05-03T20:58:36.158+0000",
    "status": 404,
    "error": "Not Found",
    "message": "No message available",
    "path": "/api/projects/1/sprints/1/availability"
}
```

Reply:
profesjonalny fix usuwający sprawdzanie typów #pocotokomu

Reply:
Dodałam sobie deklarację do bazy i jest
![obraz](https://user-images.githubusercontent.com/34098778/57165438-6461ee00-6df7-11e9-9a25-7155c3300890.png)

Reply:
Ale pierwsze deklaracja też by była spoko

Reply:
ładna data przychodzi xd zastanawiałem się w sumie co przyjdzie z backendu

Reply:
aa, okej, bo w sumie nie uwzględniłem nigdzie sytuacji braku jakiejkolwiek deklaracji... no okej, poprawię

Reply:
Ja nie wiem, gdzie jest problem, to jest super format

![obraz](https://user-images.githubusercontent.com/34098778/57165525-a9862000-6df7-11e9-8e46-52d62fec10e8.png)

Reply:
a co do daty, to zostawiamy format w api jaki jest, czy może raczej string w jakimś ustalonym formacie?

Reply:
tutaj chyba nie ma potrzeby wykonywania operacji na dacie, więc może być taka bardziej _readonly_

Reply:
poprawiłem backend (chyba)
turn in declaration?

Reply:
Uwagi:
1. Mam poczucie, że ta czcionka jest większa niż czcionka w każdym innym miejscu frontu
2. Pole na komentarze większe?
3. `Zatwierdź` i `Powrót` jakoś tak odwrotnie? `Zatwierdź` bardziej z prawej czy coś?
4. `Zatwierdz` nie ma ogonka nad `ź` - mogę się zaraz zmęczyć i poszukać tego w kodzie, ale na oko nie ma

![obraz](https://user-images.githubusercontent.com/34098778/57165824-8019c400-6df8-11e9-99e1-4051541e8705.png)

Reply:
a te liczby to też jakiś bug? xd

Reply:
TYLE MINUT BĘDĘ PRACOWAĆ

Reply:
no nie ma ogonka, ale i tak planowałem trochę przerobić ten formularz (bo nie jest to formularz, a fajnie by było, żeby jednak był + ujemny czas + nie podoba mi się sposób wprowadzania tych liczb i to że tylko godziny można wprowadzać)

Reply:
Nie no, to są nanosekundy chyba

Reply:
aa i jeszcze trzeba to pewnie pomnożyć (albo uspójnić) bo na froncie jest napisane że niby w godzinach czas pracy, a w api leci w minutach

~(a programista nigdzie nie pomnożył hehe)~

Reply:
Racja, formularz mi mówi o godzinach, a ja będę pracować sześć minut

Reply:
W sumie "przepracuejesz" to chyba też nie jest słowo

Reply:
Kto czyta opisy text boxów

Reply:
KURDE NOOO xd nie sprawdzałem tego xd
poprawię kiedyś

Reply:
ważne że w miarę działa

Reply:
W sensie ja nie czytam - wpisałam sobie swoje 6 minut i byłam zadowolona

Reply:
A minuty wpisałam, BO WIEM, ŻE TAM SĄ MINUTY POD SPODEM

Reply:
Działa, działa

Reply:
a nie mów nawet, prawie dzisiaj nic nie ruszyłem z nimi 
jedyny mój sukces, to że mvn install generuje mi wszystkie interfejsy

Reply:
a jeszcze iot

Reply:
Ja jeszcze mam pół zestawu z kompilatorów na poniedziałek

Reply:
Same fajne sprawy

Reply:
Napisz jeszcze jakiś widoczek, to ja potestuję

Reply:
btw domyślna czcionka nie ma polskich znaków xd